### PR TITLE
Database Rebase and Table Rename

### PR DESCRIPTION
**Pull request by LtRipley36706**

### CODE REVIEW

**Comment by Mag-nus** [comment]
Can you summarize the changes being made here?

Also, if we're making these changes, can we move the player specific tables over to character_ ?

File: Source/ACE.Server/WorldObjects/WorldObject_Properties.cs [hunk 1]
```diff
@@ -329,14 +329,14 @@ public void RemovePosition(PositionType positionType)
         {
             Positions.Remove(positionType);
 
-            if (Biota.TryRemovePosition(positionType, out var entity, biotaPropertiesPositionLock) && ExistsInDatabase && entity.Id != 0)
+            if (Biota.TryRemovePosition(positionType, out var entity, biotaPropertiesPositionLock) && ExistsInDatabase && entity.ObjectId != 0)
                 DatabaseManager.Shard.RemoveEntity(entity, null);
         }
```
Comment (Mag-nus) [inline Source/ACE.Server/WorldObjects/WorldObject_Properties.cs]: If we remove the auto generated id and now rely on the biota's id as the primary key for the record, this code will break.

entity.Objectid will always be non-zero as it's generated by ACE to put the object into the world.

entity.Id would be non-zero until it's saved to the db. At that point the db gives it a unique record id.

**Comment by Mag-nus** [comment]
See the comment I made on WorldObject_Properties.cs

**Comment by LtRipley36706** [comment]
This PR resolves #782 and resolves #783

File: Database/Base/WorldBase.sql [hunk 2]
```diff
@@ -102,6 +153,456 @@ CREATE TABLE `points_of_interest` (
...
+/*!40101 SET character_set_client = @saved_cs_client */;
+
+--
+-- Table structure for table `spell`
+--
+
+DROP TABLE IF EXISTS `spell`;
+/*!40101 SET @saved_cs_client     = @@character_set_client */;
+/*!40101 SET character_set_client = utf8 */;
+CREATE TABLE `spell` (
```
Comment (Mag-nus) [inline Database/Base/WorldBase.sql]: In this table, was the duplicate data that exists in the dat removed from the db?

There's no need to duplicate the data, and, in fact, we must not. The client will be pulling from the dat for those values and the server will need to as well.

File: Source/ACE.Database/Models/Shard/Character.cs [hunk 3]
```diff
@@ -12,6 +12,11 @@ public partial class Character
         public bool IsDeleted { get; set; }
         public ulong DeleteTime { get; set; }
         public double LastLoginTimestamp { get; set; }
+        public int TotalLogins { get; set; }
+        public int CharacterOptions1 { get; set; }
+        public int CharacterOptions2 { get; set; }
+        public uint HairTexture { get; set; }
+        public uint DefaultHairTexture { get; set; }
```
Comment (Mag-nus) [inline Source/ACE.Database/Models/Shard/Character.cs]: This looks good.

File: Database/Base/WorldBase.sql (hunk 2 above)
Comment (LtRipley36706) [reply Database/Base/WorldBase.sql]: I made no changes here. before removing the data, I would need know exactly what columns are duplicated and it would also make sense to note what differences, if any, there are to the exported data vs DAT data. I think the coding itself uses the combined data, starting with the DAT then moving over to the DB to fill in the rest but I'm not as well versed in this part as @kdmxp256 or maybe @gmriggs
